app/models/users/suspended.rb

```diff
@@ -0,0 +1,20 @@
+module Users
+  class Suspended < ApplicationRecord
+    self.table_name = "suspended_users"
+
+    validates :username_hash, presence: true, uniqueness: true
+
+    def self.hash_username(username)
+      Digest::SHA256.hexdigest(username)
+    end
+
+    def self.previously_banned?(username)
+      where(username_hash: hash_username(username)).any?
+    end
+
+    # Convenience method for easily adding a suspended user
+    def self.add(user)
```

Comment:
[Suggestion] Consider renaming this method to `create` to match ActiveRecord conventions, and allow usage like `User:Suspended.create(user)`

Reply:
They are semantically different though: `create` expects an attribute hash, not a user object. I think violating this would break `ActiveRecord` conventions more than adding a "factory method" like this.

Reply:
I may call it `create_from_user` though, that has a nice ring to it and alludes to `create` being called under the hood. Thanks for making me rethink the name 😃

Reply:
Oh, I quite like `create_from_user`, and point taken about the attribute hash!